pkg/asset/agent/joiner/clusterinfo.go

```diff
@@ -94,10 +95,17 @@ func (ci *ClusterInfo) Generate(dependencies asset.Parents) error {
 	if err != nil {
 		return err
 	}
-	err = ci.retrieveArchitecture()
-	if err != nil {
-		return err
+
+	if addNodesConfig.Config.CPUArchitecture != "" {
```

Comment:
Can you please move this code inside the `ci.retrieveArchitecture()` method? It's logically part of that action, and it would help in keeping more readable the `Generate` method